assets/scripts/app/__tests__/NoConnectionMessage.test.js

```diff
@@ -1,44 +1,51 @@
 /* eslint-env jest */
 import React from 'react'
-import { shallow } from 'enzyme'
-import { mountWithIntl as mount } from '../../../../test/helpers/intl-enzyme-test-helper.js'
-import { NoConnectionMessage } from '../NoConnectionMessage'
+import { fireEvent } from '@testing-library/react'
+import { renderWithReduxAndIntl } from '../../../../test/helpers/render'
+import NoConnectionMessage from '../NoConnectionMessage'
 import { nonblockingAjaxTryAgain } from '../../util/fetch_nonblocking'
+import { showNoConnectionMessage } from '../../store/actions/status'
 
 jest.mock('../../util/fetch_nonblocking', () => ({
   nonblockingAjaxTryAgain: jest.fn()
 }))
 
-describe('NoConnectionMessage', () => {
-  it('renders a button', () => {
-    const wrapper = shallow(<NoConnectionMessage />)
-    expect(wrapper.find('button').length).toEqual(1)
-  })
+const initialState = {
+  status: {
+    noConnectionMessage: false
+  }
+}
 
+describe('NoConnectionMessage', () => {
   it('tries to reconnect when button is clicked', () => {
-    const wrapper = shallow(<NoConnectionMessage />)
-    wrapper.find('button').simulate('click')
+    const wrapper = renderWithReduxAndIntl(<NoConnectionMessage />, {
+      initialState
+    })
+    fireEvent.click(wrapper.getByText('Try again'))
     expect(nonblockingAjaxTryAgain).toHaveBeenCalled()
   })
 
   it('does not have the visibility class when mounted', () => {
-    const wrapper = shallow(<NoConnectionMessage />)
-    expect(wrapper.find('.status-message-visible').length).toEqual(0)
+    const wrapper = renderWithReduxAndIntl(<NoConnectionMessage />, {
+      initialState
+    })
+    expect(wrapper.asFragment()).toMatchSnapshot()
```

Comment:
Since finding DOM elements is testing implementation detail, we no longer look for elements by class name. So this test has been modified instead to render a snapshot. A snapshot lets us compare the rendered HTML so we can see if a change in the code results in the expected HTML change or not. We can also use this snapshot to verify that the class names of our element is what we expect.

The component could be updated to render other indicators of visibility, but in this example, we are porting only the test suite and not the original component.